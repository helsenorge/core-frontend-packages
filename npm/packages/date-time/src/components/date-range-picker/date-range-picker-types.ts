import {
  FocusedInputShape,
  SingleDatePickerPhrases,
  DateRangePickerPhrases,
  SingleDatePickerShape,
  DateRangePickerShape,
} from 'react-dates';

import { LanguageLocales } from '@helsenorge/core-utils/constants/languages';

export type DatePickerErrorPhrases = {
  errorInvalidDate: string;
  errorInvalidDateRange: string;
  errorRequiredDate: string;
  errorRequiredDateRange: string;
  errorInvalidMinimumNights: string;
  errorAfterMaxDate: string;
  errorBeforeMinDate: string;
};
export type SingleDatePickerNode = React.ClassicComponent<SingleDatePickerShape, {}> & {
  container?: HTMLElement;
};
export type DateRangePickerNode = React.ClassicComponent<DateRangePickerShape, {}> & {
  container?: HTMLElement;
};

export interface DateRangePickerDefaultProps {
  /** Unik ID som sendes videre til komponenten - spesifikk til type === 'single'  */
  id: string;
  /** Unik ID som sendes videre til startDate inputfelt - spesifikk til type === 'range'  */
  startDateId: string;
  /** Placeholder som brukes i startDate input feltet - spesifikk til type === 'range' - default er 'Startdato' */
  startDatePlaceholderText: string;
  /** Unik ID som sendes videre til endDate inputfelt - spesifikk til type === 'range'  */
  endDateId: string;
  /** Placeholder som brukes i endDate input feltet - spesifikk til type === 'range' - default er 'Sluttdato' */
  endDatePlaceholderText: string;
  /** Forhåndsdefinerte string Resources som konsumeres av komponenten - kan overskrives */
  resources: SingleDatePickerPhrases | DateRangePickerPhrases;
  /** Forhåndsdefinerte error string Resources som konsumeres av komponenten - kan overskrives */
  errorResources: DatePickerErrorPhrases;
  /** Viser kalender icon i inputfeltet - default er true */
  showDefaultInputIcon: boolean;
  /** Antall måneder som skal vises i kalenderpanelet - default er 1 */
  numberOfMonths: number;
  /* Dersom mol_validation og error meldinger ved validering skal skjules */
  isValidationHidden: boolean;
  /** Definerer om feltet er required - brukes både for validering og for styling */
  isRequired: boolean;
  /** Definerer om feltet er disabled */
  isDisabled: boolean;
}

export interface Props {
  /** Definerer om det skal velges én dato eller en periode  */
  type: 'single' | 'range';
  /** Definerer hvilken locale moment bruker - default en 'nb-no'  */
  locale?: LanguageLocales.NORWEGIAN | LanguageLocales.ENGLISH | LanguageLocales.SAMI_NORTHERN;
  /** Optional label som settes over datepickeren */
  label?: string | JSX.Element;
  /** Teksten til sub label, brukes som enkel hjelpetekst. Sublabel legges som et eget blokk-element nederst i <legend> til dette feltet */
  subLabel?: string | JSX.Element;
  /** Placeholder som brukes i input feltet - spesifikk til type === 'single'  - default er 'dd.mm.åååå' */
  placeholder?: string;
  /** Custom className som settes på wrapperen  */
  className?: string;
  /** Verdien som settes som default på DatePickeren når den initieres */
  singleDateValue?: moment.Moment;
  /** Verdien som settes som default på startDate i Daterangepickeren når den initieres */
  startDateValue?: moment.Moment;
  /** Verdien som settes som default på startDate i Daterangepickeren når den initieres */
  endDateValue?: moment.Moment;
  /** Minimum dato som kan velges. Brukes både for visning i kalenderen og for validering */
  minimumDate?: moment.Moment;
  /** Maksimal dato som kan velges. Brukes både for visning i kalenderen og for validering*/
  maximumDate?: moment.Moment;
  /** Definerer minimum antall dager for en periode - spesifikk til type === 'range'   */
  minimumPeriod?: number;
  /** Dato visningen i kalenderen starter i. Brukes kun når minimumDate ikke er i bruk. default er faktisk value, og 'now' */
  initialDate?: moment.Moment;
  /** Optional function som kalles når datoen endrer seg. Hvis type === range sendes det et objekt med begge datoer som andre parameter */
  onDateChange?: (value: moment.Moment | { start: moment.Moment | null; end: moment.Moment | null } | null, id?: string) => void;
  /* Teksten til required label */
  requiredLabel?: string;
  /* Teksten til optional label - krever at required er false for å vises */
  optionalLabel?: string;
  /** Setter en --hidden className på label - fortsatt lesbart for screen-readers men ikke synlig */
  isLabelHidden?: boolean;
  /** Definerer om input feltet skal ta hele bredden - gjelder både mobil og desktop - default er false  */
  hasFullWidth?: boolean;
  /** Definerer om måned og år skal vise en enkel string istedenfor dropdowns: default er false  */
  isMonthHeaderSimplified?: boolean;
  /** Optional JSX.Element som vises ved error istedenfor standard ValidationError fra Form */
  validationErrorRenderer?: JSX.Element;
  /** Ekstra validator check - returnerer om verdien er gyldig og errorString som skal vises  */
  dateValidator?: (id: string, value?: moment.Moment, endDate?: moment.Moment) => { isValid: boolean; errorString?: string };
  /** Function som kalles når en dato er feil */
  onError?: (date: moment.Moment | { start: moment.Moment | null; end: moment.Moment | null } | null, error?: string, id?: string) => void;
  /** Function som kalles når datoen valideres riktig */
  onValidated?: (isValid: boolean) => void;
  /** Hjelpetrigger som vises etter label */
  helpButton?: JSX.Element;
  /** Element som vises når man klikker på HelpButton */
  helpElement?: JSX.Element;
}

export type DateRangePickerProps = DateRangePickerDefaultProps & Props;

export type PartialPropsForDesktop = Omit<
  DateRangePickerProps,
  | 'locale'
  | 'className'
  | 'label'
  | 'isLabelHidden'
  | 'requiredLabel'
  | 'optionalLabel'
  | 'sublabel'
  | 'validationErrorRenderer'
  | 'isValidationHidden'
  | 'helpButton'
  | 'helpElement'
>;

export interface DateRangePickerState {
  isMobile: boolean;
  momentLocale: moment.Locale;
  placeholder: string;
  singleDate: moment.Moment | null;
  isSingleDateFocused: boolean | null;
  isSingleDateValid: boolean;
  startDate: moment.Moment | null;
  endDate: moment.Moment | null;
  isRangeDateValid: boolean;
  validated: boolean;
  focusedInput: FocusedInputShape | null;
  errorString?: string;
  isNextMonthDisabled?: boolean;
  isPrevMonthDisabled?: boolean;
  isFocusReset?: boolean;
  visibleMonth: moment.Moment;
}
