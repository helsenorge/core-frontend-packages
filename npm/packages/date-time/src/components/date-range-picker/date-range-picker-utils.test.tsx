import { mount } from 'enzyme';
import moment from 'moment';

import { DateRangePickerProps } from './date-range-picker-types';
import * as DateRangePickerUtils from './date-range-picker-utils';

describe('DateRangePicker date-range-picker-utils', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  describe('Gitt at renderYearsOptions kalles', () => {
    describe('Når den har min 2010 og max 2020', () => {
      const a = DateRangePickerUtils.renderYearsOptions(2010, 2020);
      it('Så har den en length på 11', () => {
        expect(a.length).toEqual(11);
      });
      it('Så består hver entry av en option med riktig value og text', () => {
        expect(JSON.stringify(a[0])).toEqual(
          '{"type":"option","key":"2010","ref":null,"props":{"value":2010,"children":2010},"_owner":null,"_store":{}}'
        );
      });
    });
  });

  describe('Gitt at renderMonthsPicker kalles', () => {
    const fnMock = jest.fn();
    describe('Når den får minDato i februar og MaxDato i november', () => {
      const a = DateRangePickerUtils.renderMonthsPicker(
        moment('20.10.2020', 'DD.MM.YYYY'),
        fnMock,
        moment().localeData(),
        moment('20.02.2020', 'DD.MM.YYYY'),
        moment('20.11.2020', 'DD.MM.YYYY')
      );

      it('Så returnerer den en dropdown med riktig className', () => {
        expect(a.props.className).toEqual('datepicker__month-selector');
        expect(a).toMatchSnapshot();
      });
      it('Så returnerer den 12 <option>, én for hver måned', () => {
        expect(a.props.children.length).toEqual(12);
      });
      it('Så har den selected verdien satt til riktig måneden (starter fra 0 pga Array index)', () => {
        expect(a.props.value).toEqual(9);
      });
      it('Så har den riktig aria-label', () => {
        expect(a.props['aria-label']).toEqual('Velg måned');
      });
    });

    describe('Når den får en minDato som har lik måned som maxDato', () => {
      const a = DateRangePickerUtils.renderMonthsPicker(
        moment('20.10.2020', 'DD.MM.YYYY'),
        fnMock,
        moment().localeData(),
        moment('01.10.2020', 'DD.MM.YYYY'),
        moment('25.10.2020', 'DD.MM.YYYY')
      );

      it('Så returnerer den visnign av bare den ene måneden - uten dropdown', () => {
        expect(a.props.className).toEqual('datepicker__month-view');
        expect(a.props.children).toEqual('October');
        expect(a).toMatchSnapshot();
      });
    });

    describe('Når locale er satt til engelsk', () => {
      moment.locale('en-GB');
      const a = DateRangePickerUtils.renderMonthsPicker(
        moment('20.10.2020', 'DD.MM.YYYY'),
        fnMock,
        moment().localeData(),
        moment('20.02.2020', 'DD.MM.YYYY'),
        moment('20.11.2020', 'DD.MM.YYYY')
      );

      it('Så er språk riktig oversatt, og aria-label er på engelsk', () => {
        expect(a.props['aria-label']).toEqual('Pick month');
      });
    });
  });

  describe('Gitt at renderYearsPicker kalles', () => {
    moment.locale('nb');
    const fnMock = jest.fn();
    describe('Når min og max dato er undefined', () => {
      const a = DateRangePickerUtils.renderYearsPicker(
        moment('20.10.2015', 'DD.MM.YYYY'),
        fnMock,
        moment().localeData(),
        undefined,
        undefined
      );
      it('Så returnerer den 201 <option>, 100 år fremover og 100 år bakover', () => {
        expect(a.props.children.length).toEqual(201);
      });
      it('Så har den selected verdien satt til riktig år', () => {
        expect(a.props.value).toEqual(2015);
      });
      it('Så er aria-label riktig', () => {
        expect(a.props['aria-label']).toEqual('Velg år');
      });
    });
    describe('Når den får minDato i 2010 og MaxDato i 2020', () => {
      const a = DateRangePickerUtils.renderYearsPicker(
        moment('20.10.2015', 'DD.MM.YYYY'),
        fnMock,
        moment().localeData(),
        moment('20.02.2010', 'DD.MM.YYYY'),
        moment('20.11.2020', 'DD.MM.YYYY')
      );

      it('Så returnerer den en dropdown med riktig className', () => {
        expect(a.props.className).toEqual('datepicker__year-selector');
        expect(a).toMatchSnapshot();
      });
      it('Så returnerer den 11 <option>, én for hvert år', () => {
        expect(a.props.children.length).toEqual(11);
      });
      it('Så har den selected verdien satt til riktig år', () => {
        expect(a.props.value).toEqual(2015);
      });
    });

    describe('Når den får en minDato som har lik år som maxDato', () => {
      const a = DateRangePickerUtils.renderYearsPicker(
        moment('20.10.2020', 'DD.MM.YYYY'),
        fnMock,
        moment().localeData(),
        moment('01.10.2020', 'DD.MM.YYYY'),
        moment('25.10.2020', 'DD.MM.YYYY')
      );

      it('Så returnerer den visning av bare det ene året - uten dropdown', () => {
        expect(a.props.className).toEqual('datepicker__year-view');
        expect(a.props.children).toEqual(2020);
        expect(a).toMatchSnapshot();
      });
    });

    describe('Når locale er satt til engelsk', () => {
      moment.locale('en-GB');
      const a = DateRangePickerUtils.renderYearsPicker(
        moment('20.10.2015', 'DD.MM.YYYY'),
        fnMock,
        moment().localeData(),
        undefined,
        undefined
      );
      it('Så er språk riktig oversatt, og aria-label er på engelsk', () => {
        expect(a.props['aria-label']).toEqual('Pick year');
      });
    });
  });

  describe('Gitt at getCSSClasses kalles', () => {
    moment.locale('nb');
    describe('Når den har både hasValidation og hasErrors false', () => {
      const a = DateRangePickerUtils.getCSSClasses('baseClass', 'errorClass', false, false, 'extraClass');
      it('Så returnerer den riktig class utenom validation og error', () => {
        expect(a).toEqual('baseClass extraClass');
      });
    });

    describe('Når den har hasValidation false og hasErrors true', () => {
      const a = DateRangePickerUtils.getCSSClasses('baseClass', 'errorClass', false, true, 'extraClass');
      it('Så returnerer den riktig class med validation og error', () => {
        expect(a).toEqual('baseClass errorClass extraClass');
      });
    });

    describe('Når den har hasValidation true og hasErrors false', () => {
      const a = DateRangePickerUtils.getCSSClasses('baseClass', 'errorClass', true, false, 'extraClass');
      it('Så returnerer den riktig class med validation og error', () => {
        expect(a).toEqual('baseClass mol_validation extraClass');
      });
    });

    describe('Når den har både hasValidation og hasErrors true', () => {
      const a = DateRangePickerUtils.getCSSClasses('baseClass', 'errorClass', true, true, 'extraClass');
      it('Så returnerer den riktig class med validation og error', () => {
        expect(a).toEqual('baseClass mol_validation mol_validation--active errorClass extraClass');
      });
    });
  });

  describe('Gitt at notifyValidated kalles', () => {
    describe('Når datoen er gyldig', () => {
      it('Så kaller den callbacks methoder med riktig argumenter', () => {
        const onValidatedMock1 = jest.fn();
        const onErrorMock1 = jest.fn();
        DateRangePickerUtils.notifyValidated('id', moment('20.10.2020', 'DD.MM.YYYY'), true, '', onValidatedMock1, onErrorMock1);

        expect(onValidatedMock1).toHaveBeenCalledWith(true);
        expect(onErrorMock1).not.toHaveBeenCalled();
      });
    });

    describe('Når datoen er ikke gyldig', () => {
      it('Så kaller den callbacks methoder med riktig argumenter', () => {
        const onValidatedMock2 = jest.fn();
        const onErrorMock2 = jest.fn();
        DateRangePickerUtils.notifyValidated('id', moment('20.10.2020', 'DD.MM.YYYY'), false, 'myErrorMsg', onValidatedMock2, onErrorMock2);

        expect(onValidatedMock2).toHaveBeenCalledWith(false);
        expect(onErrorMock2).toHaveBeenCalledWith(moment('20.10.2020', 'DD.MM.YYYY'), 'myErrorMsg', 'id');
      });
    });
  });

  describe('Gitt at renderCalendarIcon kalles', () => {
    describe('Når den instansieres med onClick method og custom css class', () => {
      const a = DateRangePickerUtils.renderCalendarIcon(jest.fn(), 'customClassName');
      const wrapper = mount(a);
      it('Så renderes ikonet riktig', () => {
        expect(wrapper.render()).toMatchSnapshot();
      });
    });
  });

  describe('Gitt at renderChevronIcon kalles', () => {
    describe('Når den instansieres med "prev" direction og custom css class', () => {
      const a = DateRangePickerUtils.renderChevronIcon('prev', 'customClassName');
      const wrapper = mount(a);
      it('Så renderes ikonet riktig', () => {
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den instansieres med "next" direction og custom css class', () => {
      const a = DateRangePickerUtils.renderChevronIcon('next', 'customClassName');
      const wrapper = mount(a);
      it('Så renderes ikonet riktig', () => {
        expect(wrapper.render()).toMatchSnapshot();
      });
    });
  });

  describe('Gitt at renderMonthHeaderSimplified kalles', () => {
    describe('Når den får datoen 20.10.2015', () => {
      const a = DateRangePickerUtils.renderMonthHeaderSimplified({ month: moment('20.10.2015', 'DD.MM.YYYY') }, moment.localeData());
      it('Så returnerer den riktig string', () => {
        expect(a).toEqual('oktober 2015');
      });
    });
  });

  describe('Gitt at renderMonthHeader kalles', () => {
    describe('Når den får datoen 20.10.2015', () => {
      const a = DateRangePickerUtils.renderMonthHeader(
        { month: moment('20.10.2015', 'DD.MM.YYYY'), onMonthSelect: jest.fn(), onYearSelect: jest.fn() },
        moment.localeData(),
        moment('20.02.2010', 'DD.MM.YYYY'),
        moment('20.11.2020', 'DD.MM.YYYY')
      );
      it('Så returnerer den en komponent med riktig className og nested dropdowns (som er allerede testet dypere over)', () => {
        expect(a.props.className).toEqual('datepicker__month-year-selector');
        expect(a).toMatchSnapshot();
      });
    });
  });

  describe('Gitt at getDateNOString kalles', () => {
    describe('Når den kalles med datoen 20.10.2015', () => {
      const a = DateRangePickerUtils.getDateNOString(moment('20.10.2015', 'DD.MM.YYYY'));
      it('Så returnerer den 20.10.2015', () => {
        expect(a).toEqual('20.10.2015');
      });
    });

    describe('Når den kalles med undefined', () => {
      it('Så returnerer den tom streng', () => {
        const a = DateRangePickerUtils.getDateNOString(undefined);
        expect(a).toEqual('');
      });
    });

    describe('Når den kalles med null', () => {
      const a = DateRangePickerUtils.getDateNOString(null);
      it('Så returnerer den tom streng', () => {
        expect(a).toEqual('');
      });
    });
  });

  describe('Gitt at getDefaultVisibleMonth kalles', () => {
    describe('Når singleDateValue, startDateValue, endDateValue, initialDate og minimumDate er satt', () => {
      it('Så returnerer den måneden for singleDateValue', () => {
        const props = {
          singleDateValue: moment('01.01.2021', 'DD.MM.YYYY'),
          startDateValue: moment('01.02.2021', 'DD.MM.YYYY'),
          endDateValue: moment('01.03.2021', 'DD.MM.YYYY'),
          initialDate: moment('01.04.2021', 'DD.MM.YYYY'),
          minimumDate: moment('01.05.2021', 'DD.MM.YYYY'),
        } as DateRangePickerProps;

        const date = DateRangePickerUtils.getDefaultVisibleMonth(props);

        expect((date as moment.Moment).format('MM-YYYY')).toEqual('01-2021');
      });
    });
    describe('Når startDateValue, endDateValue, initialDate og minimumDate er satt', () => {
      it('Så returnerer den måneden for startDateValue', () => {
        const props = {
          startDateValue: moment('01.02.2021', 'DD.MM.YYYY'),
          endDateValue: moment('01.03.2021', 'DD.MM.YYYY'),
          initialDate: moment('01.04.2021', 'DD.MM.YYYY'),
          minimumDate: moment('01.05.2021', 'DD.MM.YYYY'),
        } as DateRangePickerProps;

        const date = DateRangePickerUtils.getDefaultVisibleMonth(props);

        expect((date as moment.Moment).format('MM-YYYY')).toEqual('02-2021');
      });
    });
    describe('Når endDateValue, initialDate og minimumDate er satt', () => {
      it('Så returnerer den måneden for endDateValue', () => {
        const props = {
          endDateValue: moment('01.03.2021', 'DD.MM.YYYY'),
          initialDate: moment('01.04.2021', 'DD.MM.YYYY'),
          minimumDate: moment('01.05.2021', 'DD.MM.YYYY'),
        } as DateRangePickerProps;

        const date = DateRangePickerUtils.getDefaultVisibleMonth(props);

        expect((date as moment.Moment).format('MM-YYYY')).toEqual('03-2021');
      });
    });
    describe('Når initialDate og minimumDate er satt', () => {
      it('Så returnerer den måneden for initialDate', () => {
        const props = {
          initialDate: moment('01.04.2021', 'DD.MM.YYYY'),
          minimumDate: moment('01.05.2021', 'DD.MM.YYYY'),
        } as DateRangePickerProps;

        const date = DateRangePickerUtils.getDefaultVisibleMonth(props);

        expect((date as moment.Moment).format('MM-YYYY')).toEqual('04-2021');
      });
    });
    describe('Når minimumDate er satt', () => {
      it('Så returnerer den måneden for minimumDate', () => {
        const props = {
          minimumDate: moment('01.05.2021', 'DD.MM.YYYY'),
        } as DateRangePickerProps;

        const date = DateRangePickerUtils.getDefaultVisibleMonth(props);

        expect((date as moment.Moment).format('MM-YYYY')).toEqual('05-2021');
      });
    });
    describe('Når ingen av singleDateValue, startDateValue, endDateValue, initialDate og minimumDate er satt', () => {
      it('Så returnerer den nåværende måned', () => {
        const originalDateNow = Date.now;
        Date.now = jest.fn(() => new Date('2050-12-01T12:13:14.000Z').getTime());
        const props = {} as DateRangePickerProps;

        const date = DateRangePickerUtils.getDefaultVisibleMonth(props);

        expect((date as moment.Moment).format('MM-YYYY')).toEqual('12-2050');

        Date.now = originalDateNow;
      });
    });
  });
});
