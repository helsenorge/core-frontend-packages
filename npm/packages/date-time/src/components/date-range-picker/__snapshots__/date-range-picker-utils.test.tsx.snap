// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker date-range-picker-utils Gitt at renderCalendarIcon kalles Når den instansieres med onClick method og custom css class Så renderes ikonet riktig 1`] = `
<span>
  <svg
    aria-hidden="true"
    class="hnds-style-icon customClassName"
    data-analyticsid="icon"
    fill="black"
    height="36"
    role="img"
    style="min-width: 36px; min-height: 36px;"
    viewBox="0 0 48 48"
    width="36"
  >
    <path
      d="M32.86 22.986a.984.984 0 010 1.967.984.984 0 010-1.967zm0 3.03a2.049 2.049 0 002.046-2.047 2.049 2.049 0 00-2.046-2.046 2.049 2.049 0 00-2.046 2.046c0 1.13.918 2.047 2.046 2.047zm6.21-7.864H8.68v-6.194h3.846v1.161a2.243 2.243 0 002.241 2.241 2.243 2.243 0 002.241-2.24v-1.162H30.62v1.161a2.244 2.244 0 002.241 2.241 2.243 2.243 0 002.241-2.24v-1.162h3.97v6.194zM8.678 35.982H39.07v-16.53H8.678v16.53zm5.148-26.273c0-.518.422-.94.941-.94.52 0 .941.422.941.94v3.41a.942.942 0 01-1.882 0V9.71zm18.093 0c0-.518.422-.94.941-.94.52 0 .941.422.941.94v3.41a.942.942 0 01-1.882 0V9.71zm3.182.95v-.95a2.243 2.243 0 00-2.24-2.24 2.244 2.244 0 00-2.242 2.24v.95H17.01v-.95a2.243 2.243 0 00-2.242-2.24 2.243 2.243 0 00-2.24 2.24v.95H7.378V37.28h32.992V10.658h-5.27zm-8.157 16.167a.984.984 0 110 1.969.984.984 0 010-1.969zm0 3.03a2.049 2.049 0 002.047-2.047 2.049 2.049 0 00-2.047-2.046 2.049 2.049 0 00-2.046 2.046c0 1.13.918 2.047 2.046 2.047z"
      fill-rule="evenodd"
    />
  </svg>
</span>
`;

exports[`DateRangePicker date-range-picker-utils Gitt at renderChevronIcon kalles Når den instansieres med "next" direction og custom css class Så renderes ikonet riktig 1`] = `
<span
  class="DayPickerNavigation_rightButton__horizontalDefault customClassName"
  tabindex="0"
>
  <svg
    aria-hidden="true"
    class="hnds-style-icon "
    data-analyticsid="icon"
    fill="#126F87"
    height="36"
    role="img"
    style="min-width: 36px; min-height: 36px;"
    viewBox="0 0 48 48"
    width="36"
  >
    <path
      d="M19.171 32.951l1.409 1.42L31.031 24 20.58 13.63l-1.409 1.42 9.02 8.95z"
    />
  </svg>
</span>
`;

exports[`DateRangePicker date-range-picker-utils Gitt at renderChevronIcon kalles Når den instansieres med "prev" direction og custom css class Så renderes ikonet riktig 1`] = `
<span
  class="DayPickerNavigation_leftButton__horizontalDefault customClassName"
  tabindex="0"
>
  <svg
    aria-hidden="true"
    class="hnds-style-icon "
    data-analyticsid="icon"
    fill="#126F87"
    height="36"
    role="img"
    style="min-width: 36px; min-height: 36px;"
    viewBox="0 0 48 48"
    width="36"
  >
    <path
      d="M30.66 32.951l-1.409 1.42L18.8 24l10.451-10.37 1.409 1.42L21.64 24z"
    />
  </svg>
</span>
`;

exports[`DateRangePicker date-range-picker-utils Gitt at renderMonthHeader kalles Når den får datoen 20.10.2015 Så returnerer den en komponent med riktig className og nested dropdowns (som er allerede testet dypere over) 1`] = `
<div
  className="datepicker__month-year-selector"
>
  <select
    aria-label="Velg måned"
    className="datepicker__month-selector"
    onChange={[Function]}
    onKeyDown={[Function]}
    value={9}
  >
    <option
      value={0}
    >
      januar
    </option>
    <option
      value={1}
    >
      februar
    </option>
    <option
      value={2}
    >
      mars
    </option>
    <option
      value={3}
    >
      april
    </option>
    <option
      value={4}
    >
      mai
    </option>
    <option
      value={5}
    >
      juni
    </option>
    <option
      value={6}
    >
      juli
    </option>
    <option
      value={7}
    >
      august
    </option>
    <option
      value={8}
    >
      september
    </option>
    <option
      value={9}
    >
      oktober
    </option>
    <option
      value={10}
    >
      november
    </option>
    <option
      value={11}
    >
      desember
    </option>
  </select>
  <select
    aria-label="Velg år"
    className="datepicker__year-selector"
    onChange={[Function]}
    onKeyDown={[Function]}
    value={2015}
  >
    <option
      value={2010}
    >
      2010
    </option>
    <option
      value={2011}
    >
      2011
    </option>
    <option
      value={2012}
    >
      2012
    </option>
    <option
      value={2013}
    >
      2013
    </option>
    <option
      value={2014}
    >
      2014
    </option>
    <option
      value={2015}
    >
      2015
    </option>
    <option
      value={2016}
    >
      2016
    </option>
    <option
      value={2017}
    >
      2017
    </option>
    <option
      value={2018}
    >
      2018
    </option>
    <option
      value={2019}
    >
      2019
    </option>
    <option
      value={2020}
    >
      2020
    </option>
  </select>
</div>
`;

exports[`DateRangePicker date-range-picker-utils Gitt at renderMonthsPicker kalles Når den får en minDato som har lik måned som maxDato Så returnerer den visnign av bare den ene måneden - uten dropdown 1`] = `
<span
  className="datepicker__month-view"
>
  October
</span>
`;

exports[`DateRangePicker date-range-picker-utils Gitt at renderMonthsPicker kalles Når den får minDato i februar og MaxDato i november Så returnerer den en dropdown med riktig className 1`] = `
<select
  aria-label="Velg måned"
  className="datepicker__month-selector"
  onChange={[Function]}
  onKeyDown={[Function]}
  value={9}
>
  <option
    value={0}
  >
    January
  </option>
  <option
    value={1}
  >
    February
  </option>
  <option
    value={2}
  >
    March
  </option>
  <option
    value={3}
  >
    April
  </option>
  <option
    value={4}
  >
    May
  </option>
  <option
    value={5}
  >
    June
  </option>
  <option
    value={6}
  >
    July
  </option>
  <option
    value={7}
  >
    August
  </option>
  <option
    value={8}
  >
    September
  </option>
  <option
    value={9}
  >
    October
  </option>
  <option
    value={10}
  >
    November
  </option>
  <option
    value={11}
  >
    December
  </option>
</select>
`;

exports[`DateRangePicker date-range-picker-utils Gitt at renderYearsPicker kalles Når den får en minDato som har lik år som maxDato Så returnerer den visning av bare det ene året - uten dropdown 1`] = `
<span
  className="datepicker__year-view"
>
  2020
</span>
`;

exports[`DateRangePicker date-range-picker-utils Gitt at renderYearsPicker kalles Når den får minDato i 2010 og MaxDato i 2020 Så returnerer den en dropdown med riktig className 1`] = `
<select
  aria-label="Velg år"
  className="datepicker__year-selector"
  onChange={[Function]}
  onKeyDown={[Function]}
  value={2015}
>
  <option
    value={2010}
  >
    2010
  </option>
  <option
    value={2011}
  >
    2011
  </option>
  <option
    value={2012}
  >
    2012
  </option>
  <option
    value={2013}
  >
    2013
  </option>
  <option
    value={2014}
  >
    2014
  </option>
  <option
    value={2015}
  >
    2015
  </option>
  <option
    value={2016}
  >
    2016
  </option>
  <option
    value={2017}
  >
    2017
  </option>
  <option
    value={2018}
  >
    2018
  </option>
  <option
    value={2019}
  >
    2019
  </option>
  <option
    value={2020}
  >
    2020
  </option>
</select>
`;
